app/presenters/publishing_api/html_attachment_presenter.rb

```diff
@@ -56,12 +56,20 @@ def base_path
     end
 
     def details
-      {
+      details_hash = {
         body: body,
         public_timestamp: public_timestamp,
         first_published_version: first_published_version?,
         political: political?,
       }
+
+      maybe_add_national_applicability(details_hash)
+    end
+
+    def maybe_add_national_applicability(content)
+      return content unless item.attachable.try(:nation_inapplicabilities)&.any?
```

Comment:
I think we should be able to use the `all_nation_applicability_selected?` method here, which might be nicer for readability rather than checking for the presence of associated `nation_inapplicabilities`.

`all_nation_applicability_selected?` [should be safe](https://github.com/alphagov/whitehall/blob/41acceca56b077a028873e3e8ca8fb545d52ffcd/app/models/edition.rb#L569-L571) to call on every type of Edition. It's [used in the edit form](https://github.com/alphagov/whitehall/blob/41acceca56b077a028873e3e8ca8fb545d52ffcd/app/views/admin/editions/_nation_fields.html.erb#L5) to drive that checkbox, and should always be correct according to [the validation rules](https://github.com/alphagov/whitehall/blob/d33a442db01b8c705f87c52b4ad8395dbc7b306a/app/models/edition/national_applicability.rb#L21-L28).

```suggestion
      return content if item.attachable.all_nation_applicability_selected?
```

Reply:
It's not always editions tried in so we will still need try but this is def nicer 👍

Reply:
Ah – this might not work then... 🤔

If a non-Edition model is passed in, I guess it'll skip past that guard clause now that it's checking for the presence of `all_nation_applicability_selected?`. So it'll hit `item.attachable.national_applicability` in the line below.